usng4j-impl/src/main/java/org/codice/usng4j/impl/CoordinateSystemTranslatorImpl.java

```diff
@@ -552,24 +590,85 @@ BoundingBox toBoundingBox(final UtmCoordinate utmCoordinate, final Integer accur
-    if (utmCoordinate.getNSIndicator() == NSIndicator.SOUTH) {
-      UtmCoordinate newUtm =
-          new UtmCoordinateImpl(
-              utmCoordinate.getZoneNumber(),
-              utmCoordinate.getEasting(),
-              utmCoordinate.getNorthing() - NORTHING_OFFSET);
-      return toLatLonNsNormalized(newUtm);
+  public DecimalDegreesCoordinate toLatLon(final UtmUpsCoordinate utmUpsCoordinate) {
+    if (utmUpsCoordinate.isUTM()) {
+      return toLatLon((UtmCoordinate) utmUpsCoordinate);
+    } else {
+      return toLatLon((UpsCoordinate) utmUpsCoordinate);
+    }
+  }
+
+  @Override
+  public DecimalDegreesCoordinate toLatLon(final UpsCoordinate upsCoordinate) {
+    final double northing = upsCoordinate.getNorthing() - FALSE_UPS_NORTHING;
+    final double easting = upsCoordinate.getEasting() - FALSE_UPS_EASTING;
+
+    final boolean isNorth = upsCoordinate.getLatitudeBand() >= 'Y';
+
+    final double lat;
+    if (northing == 0.0 && easting == 0.0) {
+      lat = isNorth ? 90.0 : -90.0;
     } else {
-      return toLatLonNsNormalized(utmCoordinate);
+      final double rho = Math.hypot(easting, northing);
+      final double t = rho != 0.0 ? rho / RHO_ADJUSTER_VALUE : Math.pow(EPSILON, 2.0);
+      final double taup = (1.0 / t - t) / 2.0;
+      final double tau = tauf(taup);
+      lat = (isNorth ? 1 : -1) * Math.atan(tau) * RAD_2_DEG;
     }
+
+    final double lon = Math.atan2(easting, isNorth ? -northing : northing) * RAD_2_DEG;
+    return new DecimalDegreesCoordinateImpl(lat, lon);
   }
 
-  private DecimalDegreesCoordinate toLatLonNsNormalized(UtmCoordinate utmCoordinate) {
+  @Override
+  public DecimalDegreesCoordinate toLatLon(final UtmCoordinate utmCoordinate) {
+    return utmToLatLonNsNormalized(utmCoordinate);
+  }
+
+  private static double atanh(final double x) {
+    return Math.log((1.0 + x) / (1.0 - x)) / 2.0;
+  }
+
+  private static double eatanhe(final double x) {
+    return ES * atanh(ES * x);
+  }
+
+  private static double taupf(final double tauValue) {
+    final double tau1 = Math.hypot(1.0, tauValue);
+    final double sig = Math.sinh(eatanhe(tauValue / tau1));
+    return Math.hypot(1.0, sig) * tauValue - sig * tau1;
+  }
+
+  private static double tauf(final double taupValue) {
+    final double e2m = 1.0 - Math.pow(ES, 2.0);
+    // To lowest order in e^2, taup = (1 - e^2) * tau = _e2m * tau; so use
+    // tau = taup/_e2m as a starting guess.  (This starting guess is the
+    // geocentric latitude which, to first order in the flattening, is equal
+    // to the conformal latitude.)  Only 1 iteration is needed for |lat| <
+    // 3.35 deg, otherwise 2 iterations are needed.  If, instead, tau = taup
+    // is used the mean number of iterations increases to 1.99 (2 iterations
+    // are needed except near tau = 0).
+    double tau = taupValue / e2m;
+    final double stol = Math.sqrt(EPSILON) / 10.0 * Math.max(1.0, Math.abs(taupValue));
```

Comment:
✏️ I know that I prefer explicitly declaring operation order with parens rather than needing to remember precedence rules.